viewing_party/party.py

```diff
@@ -0,0 +1,196 @@
+                else:
+                  genre_dict[val] +=1
+    
+  counter = 0
+  genree = None
+  for key, val in genre_dict.items():
+    if val > counter:
+      counter = val
+      genree = key 
+  return genree
+    
+def get_unique_watched(amandas_data):
+
+    set_watched = set()
+    set_friends = set()
+
+    for each in amandas_data["watched"]:
+        set_watched.add(each["title"])
+  
+    
+    for each_friends in amandas_data["friends"]:
+        inner_list_values = each_friends["watched"]  # this is dictionaries collected as a list as values of a key ("watched")
+        for j in range(len(inner_list_values)):   
+            set_friends.add(inner_list_values[j]["title"])
+
+    unique_set  = set_watched - set_friends
+    new_unique_list = list(unique_set)
+    print(unique_set)
+    return_list = []
+    
+    
+    for j in range(len(new_unique_list)):
+        unique_dictionaries = {}
+        unique_dictionaries["title"] = new_unique_list[j]
+        return_list.append(unique_dictionaries)
+
+    return return_list
+
+def get_friends_unique_watched(amandas_data):
+    set_watched = set()
+    set_friends = set()
+
+    for each in amandas_data["watched"]:
+        set_watched.add(each["title"])
+ 
+    
+    for each_friends in amandas_data["friends"]:
+        inner_list_values = each_friends["watched"]  # this is dictionaries collected as a list as values of a key ("watched")
+        for j in range(len(inner_list_values)):   
+            set_friends.add(inner_list_values[j]["title"])
+
+    unique_set  = set_friends - set_watched 
+    new_unique_list = list(unique_set)
+    
+    return_list = []
+    
+    
+    for j in range(len(new_unique_list)):
+        unique_dictionaries = {}
+        unique_dictionaries["title"] = new_unique_list[j]
+        return_list.append(unique_dictionaries)
+
+    return return_list
+
+
+def get_available_recs(amandas_data):
+
+    set_watched = set()
+    set_friends = set()
+    set_hosts = set()
+    title_host_dict = {}
+    movie_recommend = []
+
+    for each in amandas_data["watched"]: # find out set of titles the user watched
+        set_watched.add(each["title"])
+  
+    
+    for each_friends in amandas_data["friends"]:
+        inner_list_values = each_friends["watched"]  # this is dictionaries collected as a list as values of a key ("watched")
+        for j in range(len(inner_list_values)):   
+            set_friends.add(inner_list_values[j]["title"])
+            title_host_dict[inner_list_values[j]["title"]] = inner_list_values[j]["host"]
+            set_hosts.add(inner_list_values[j]["host"])
+
+    list_subscriptions = amandas_data["subscriptions"]
+   
+    for each_set in set_friends:
+        if  each_set not in set_watched and title_host_dict[each_set] in list_subscriptions:
+            movie_recommend.append ({"title" : each_set, "host" : title_host_dict[each_set]})
+   
+    return movie_recommend
+
+
+
+
+def get_new_rec_by_genre(sonyas_data):
```

Comment:
I am having a bit of trouble reading this function. The function should return a movie that is the genre of the `most_watched_genre`, but I don't see where the `most_watched_genre` is found here. I would expect a call to `get_most_watched_genre`. I think a few more comments and more meaningful variable names could enhance readability here.